test/local/helpers.py

```diff
@@ -191,3 +200,31 @@ def stop_process(process):
 	except subprocess.TimeoutExpired:
 		process.kill()
 		process.wait()
+
+
+def _send_external_icmp_echo(dst_ipv4, ul_ipv6):
+	icmp_pkt = (Ether(dst=PF0.mac, src=ipv6_multicast_mac, type=0x86DD) /
+			    IPv6(dst=ul_ipv6, src=router_ul_ipv6, nh=4) /
+			    IP(dst=dst_ipv4, src=public_ip) /
+			    ICMP(type=8, id=0x0040))
+	delayed_sendp(icmp_pkt, PF0.tap)
+
+def external_ping(dst_ipv4, ul_ipv6):
+	threading.Thread(target=_send_external_icmp_echo, args=(dst_ipv4, ul_ipv6)).start()
+	answer = sniff_packet(PF0.tap, is_icmp_pkt, skip=1)
+	assert answer[ICMP].code == 0, \
+		"Invalid ICMP echo response"
+
+
+def _send_external_icmp_echo6(dst_ipv6, ul_ipv6):
+	icmp_pkt = (Ether(dst=PF0.mac, src=ipv6_multicast_mac, type=0x86DD) /
+				IPv6(dst=ul_ipv6, src=router_ul_ipv6, nh=0x29) /
+				IPv6(dst=dst_ipv6, src=public_ipv6, nh=58) /
+				ICMPv6EchoRequest())
+	delayed_sendp(icmp_pkt, PF0.tap)
+
+def external_ping6(dst_ipv6, ul_ipv6):
```

Comment:
Functionally this is pure refactoring, but the reason to do this was because I need to encapsulate all `sniff()` calls to intercept the mangled packets with EtherType=1337 and fix them.
These calls were the last three instances where a "naked" scapy call was being made.